minEnclosingTriangle() crashes when called with certain points

##### System information (version)
- OpenCV => 3.4.1
- Operating System / Platform => Linux Mint 18.1 64-bit
- Compiler => GCC 5.4.0 

##### Detailed description

The function minEnclosingTriangle crashes, when called with the points in the example below.
```
Error Message:
terminate called after throwing an instance of 'cv::Exception'
what(): OpenCV(3.4.1-dev) .../imgproc/src/min_enclosing_triangle.cpp:629: error: (-3) Internal error: The position of side B could not be determined, because gamma(b) could not be computed. in function updateSideB
```

##### Steps to reproduce
Example Program for Error reproduction: 
```.cpp
#include <iostream>
#include <opencv2/imgproc.hpp>

int main() {
  std::vector<cv::Point2f> points, triangle;

  points = std::vector<cv::Point2f>({cv::Point2f(0.40918755531311035, -0.3420603275299072),
                                     cv::Point2f(0.4073622226715088, -0.33848732709884644),
                                     cv::Point2f(0.3941684663295746, -0.32085341215133667),
                                     cv::Point2f(0.3630530536174774, -0.2843431830406189),
                                     cv::Point2f(0.3289664387702942, -0.24579767882823944),
                                     cv::Point2f(0.2949128746986389, -0.20906798541545868),
                                     cv::Point2f(0.2928275763988495, -0.2069760113954544),
                                     cv::Point2f(0.2849937379360199, -0.2016451358795166),
                                     cv::Point2f(0.2739992141723633, -0.1981959044933319),
                                     cv::Point2f(0.2711915373802185, -0.1987496167421341)
                                    });

  cv::minEnclosingTriangle(points, triangle);

  std::cout << "Triangle:" << std::endl;
  std::cout << triangle[0] << std::endl;
  std::cout << triangle[1] << std::endl;
  std::cout << triangle[2] << std::endl;

  return 0;
}
```

--- a/modules/imgproc/src/min_enclosing_triangle.cpp
+++ b/modules/imgproc/src/min_enclosing_triangle.cpp
@@ -988,14 +988,15 @@ static bool findGammaIntersectionPoints(const std::vector<cv::Point2f> &polygon,
     double sideCExtraParam = 2 * polygonPointHeight * distFormulaDenom;
 
     // Get intersection points if they exist or if lines are identical
-    if (!areIntersectingLines(side1Params, side2Params, sideCExtraParam, intersectionPoint1, intersectionPoint2)) {
-        return false;
+    if (areIntersectingLines(side1Params, side2Params, sideCExtraParam, intersectionPoint1, intersectionPoint2)) {
+        return true;
     } else if (areIdenticalLines(side1Params, side2Params, sideCExtraParam)) {
         intersectionPoint1 = side1StartVertex;
         intersectionPoint2 = side1EndVertex;
+        return true;
     }
 
-    return true;
+    return false;
 }
 
 //! Check if the given lines are identical or not

